modules/ssp_genieeBidAdapter.js

```diff
@@ -0,0 +1,449 @@
+    data.lati = encodeURIComponentIncludeSingleQuotation(
+      geparameter[GEPARAMS_KEY.LATITUDE]
+    );
+  }
+  if (GEPARAMS_KEY.CUSTOM in geparameter && isObject(geparameter[GEPARAMS_KEY.CUSTOM])) {
+    for (const c in geparameter[GEPARAMS_KEY.CUSTOM]) {
+      if (hasParamsNotBlankString(geparameter[GEPARAMS_KEY.CUSTOM], c)) {
+        data[encodeURIComponentIncludeSingleQuotation('custom_' + c)] =
+          encodeURIComponentIncludeSingleQuotation(
+            geparameter[GEPARAMS_KEY.CUSTOM][c]
+          );
+      }
+    }
+  }
+  const gecuparameter = window.gecuparams || {};
+  if (isObject(gecuparameter)) {
+    if (hasParamsNotBlankString(gecuparameter, GECUPARAMS_KEY.VERSION)) {
+      data.gc_ver = encodeURIComponentIncludeSingleQuotation(gecuparameter[GECUPARAMS_KEY.VERSION]);
+    }
+    if (hasParamsNotBlankString(gecuparameter, GECUPARAMS_KEY.MINOR_VERSION)) {
+      data.gc_minor = encodeURIComponentIncludeSingleQuotation(gecuparameter[GECUPARAMS_KEY.MINOR_VERSION]);
+    }
+    if (hasParamsNotBlankString(gecuparameter, GECUPARAMS_KEY.VALUE)) {
+      data.gc_value = encodeURIComponentIncludeSingleQuotation(gecuparameter[GECUPARAMS_KEY.VALUE]);
+    }
+  }
+
+  // imuid
+  const imuidQuery = getImuidAsQueryParameter();
+  if (imuidQuery) data.extuid = imuidQuery;
+
+  // makeUAQuery
+  // To avoid double encoding, not using encodeURIComponent here
+  const ua = JSON.parse(getUserAgent());
+  if (ua && ua.fullVersionList) {
+    const fullVersionList = ua.fullVersionList.reduce((acc, cur) => {
+      let str = acc;
+      if (str) str += ',';
+      str += '"' + cur.brand + '";v="' + cur.version + '"';
+      return str;
+    }, '');
+    data.ucfvl = fullVersionList;
+  }
+  if (ua && ua.platform) data.ucp = '"' + ua.platform + '"';
+  if (ua && ua.architecture) data.ucarch = '"' + ua.architecture + '"';
+  if (ua && ua.platformVersion) data.ucpv = '"' + ua.platformVersion + '"';
+  if (ua && ua.bitness) data.ucbit = '"' + ua.bitness + '"';
+  data.ucmbl = '?' + (ua && ua.mobile ? '1' : '0');
+  if (ua && ua.model) data.ucmdl = '"' + ua.model + '"';
+
+  return data;
+}
+
+/**
+ * making request data for banner
+ */
+function makeBannerRequestData(bid, geparameter, refererInfo) {
+  const data = makeCommonRequestData(bid, geparameter, refererInfo);
+
+  // this query is not used in nad endpoint but used in ad_call endpoint
+  if (hasParamsNotBlankString(geparameter, GEPARAMS_KEY.BUNDLE)) {
+    data.apid = encodeURIComponentIncludeSingleQuotation(geparameter[GEPARAMS_KEY.BUNDLE]);
+  }
+
+  return data;
+}
+
+/**
+ * making bid response be used commonly banner and native
+ */
+function makeCommonBidResponse(bid, width, height) {
+  return {
+    requestId: bid.requestid,
+    cpm: bid.price,
+    creativeId: bid.creativeId,
+    currency: bid.cur,
+    netRevenue: NET_REVENUE,
+    ttl: 700,
+    width: width + 'px', // width of the ad iframe
+    height: height + 'px', // height of the ad iframe
```

Comment:
width & height should be numbers

Reply:
I modify them to number style (removing ` + 'px'` ).